fdbclient/NativeAPI.actor.cpp

```diff
@@ -775,9 +779,51 @@ Future<Void> DatabaseContext::onConnected() {
 	return cluster->onConnected();
 }
 
+ACTOR static Future<Void> changeConnectionFileImpl(Reference<ClusterConnectionFile> connFile, DatabaseContext* self) {
+	TEST(true); // Change connection file
+	TraceEvent("ChangeClusterFile")
+	    .detail("ClusterFile", connFile->canGetFilename() ? connFile->getFilename() : "")
+	    .detail("ConnectionString", connFile->getConnectionString().toString());
+
+	// Reset state from former cluster.
+	self->masterProxies.clear();
+	self->minAcceptableReadVersion = std::numeric_limits<Version>::max();
+	self->invalidateCache(allKeys);
+	self->cluster->getClusterInterface()->set({});
+
+	self->cluster->getConnectionFile()->set(connFile);
+	state Database db(Reference<DatabaseContext>::addRef(self));
+	state Transaction tr(db);
+	loop {
+		tr.setOption(FDBTransactionOptions::READ_LOCK_AWARE);
+		try {
+			TraceEvent("ChangeClusterFileAttemptingGRV");
+			Version v = wait(tr.getReadVersion());
+			TraceEvent("ChangeClusterFileGotRV")
+			    .detail("ReadVersion", v)
+			    .detail("MinAcceptableReadVersion", self->minAcceptableReadVersion);
+			ASSERT(self->minAcceptableReadVersion != std::numeric_limits<Version>::max());
+			TraceEvent("ChangeClusterFileRecreateWatchesTrigger");
```

Comment:
This trace is probably not needed because it doesn't provide any non-constant information and will always show up after the previous event (or, if for some reason the assert fires, it will be clear what happened in that case as well).

Reply:
Removed.